src/task/create.ts

```diff
@@ -1,40 +1,48 @@
 import { Meta$, type Meta } from "@meta"
 import { type Eventual, type UnitName } from "@shared"
-import { literal, reference, type ReferenceProvider, type SpotContext } from "@spot"
+import { literal, reference, type ExtractSpotContext, type ReferenceProvider, type SpotContext } from "@spot"
 
 const Task$ = Symbol("$task")
 
-type ContextOfRunner<Context> = Context extends void ? void : SpotContext<Context>
+type ContextOfRunner<RunContext> = RunContext extends void ? void : SpotContext<RunContext>
 
 type TaskInternal = {
   id: { value: symbol; status: symbol }
   run: (ctx: any) => Eventual<unknown>
   enabled?: (ctx: any) => Eventual<boolean>
-  context: unknown
+  context: { run: unknown; enabled: unknown }
 }
 
 type Task<Api> = ReferenceProvider<Api> & { [Task$]: TaskInternal } & Meta
 
 type AnyTask = Task<unknown>
 
-type TaskConfig<Context, Api> = {
+type TaskConfig<RunContext, EnabledContext, Api> = {
   name: UnitName
-  run: { fn: (ctx: Context) => Eventual<Api> } & (Context extends void
+  run: { fn: (ctx: RunContext) => Eventual<Api> } & (RunContext extends void
     ? { context?: never }
-    : { context: ContextOfRunner<Context> })
-  enabled?: { fn: (ctx: Context) => Eventual<boolean> }
+    : { context: ContextOfRunner<RunContext> })
+  enabled?: EnabledContext extends void
+    ? { fn: (ctx?: never) => Eventual<boolean>; context?: never }
```

Comment:
Can we remove the `ctx` argument completely?
```suggestion
    ? { fn: () => Eventual<boolean>; context?: never }
```